app/livechat/server/lib/routing/AutoSelection.js

```diff
@@ -20,11 +21,12 @@ class AutoSelection {
 	}
 
 	getNextAgent(department, ignoreAgentId) {
+		const extraQuery = callbacks.run('livechat.applySimultaneousChatRestrictions', { ...department ? { departmentId: department } : {} });
 		if (department) {
-			return LivechatDepartmentAgents.getNextAgentForDepartment(department, ignoreAgentId);
+			return LivechatDepartmentAgents.getNextAgentForDepartment(department, ignoreAgentId, extraQuery); // this is the place where the alg selects an agent
```

Comment:
```suggestion
			return LivechatDepartmentAgents.getNextAgentForDepartment(department, ignoreAgentId, extraQuery);
```

Do we still need this comment?

Reply:
Nope, we can remove it!